src/mgmt/services/manager/project/api/nodes.py

```diff
@@ -62,6 +217,50 @@ def single_node(node_id):
         db.session.commit()
         response_object = node.to_json()
     elif request.method == 'DELETE':
+        post_data = request.get_json()
+        node_name = node.name
+        zgc_id = node.zgc_id
+        all_droplets_in_zgc = obj_api.list_cluster_custom_object(group='zeta.com',
+                                                                 version='v1',
+                                                                 plural='droplets',
+                                                                 label_selector='zgc_id='+zgc_id+',network=tenant'
+                                                                )['items']
+        droplet_to_remove = None
+
+        tenant_droplet_name = 'droplet-tenant-'+node_name.replace('_', "-").lower()
+        zgc_droplet_name = 'droplet-zgc-'+node_name.replace('_', "-").lower()
+
+        for droplet in all_droplets_in_zgc:
+            if droplet['metadata']['name'] == tenant_droplet_name:
+                droplet_to_remove = droplet
+                all_droplets_in_zgc.remove(droplet)
+                break
+
+        if droplet_to_remove is None:
+            logger.error('Cannot find that droplet with name: {}, thus not deleting anything'.format(tenant_droplet_name))
+            return jsonify({})
+        else:
+            ip_to_assign = droplet_to_remove['spec']['ip']
+
+            ip_amount = len(ip_to_assign)
+
+            number_of_droplets_to_assign = len(all_droplets_in_zgc)
+
+            modified_droplets = []
+            for index in range(ip_amount):
+                ip = ip_to_assign[index]
+                mac = get_mac_from_ip(ip)
+                all_droplets_in_zgc[index % number_of_droplets_to_assign]['spec']['ip'].append(ip)
+                all_droplets_in_zgc[index % number_of_droplets_to_assign]['spec']['mac'].append(mac)
+                if all_droplets_in_zgc[index % number_of_droplets_to_assign] not in modified_droplets: 
+                    modified_droplets.append(all_droplets_in_zgc[index % number_of_droplets_to_assign])
+
```

Comment:
dictionary may be better than list to avoid checking existence

Reply:
You're right, that was also my initial implementation and it didn't work, it is because that those droplets are already dictionaries, and in Python we cannot put dictionaries in another dictionary.

Reply:
https://www.programiz.com/python-programming/nested-dictionary